# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/EditorBase.h
# Commit: cdcef4f3f635
# Full Hash: cdcef4f3f63553a28a1055e6f4f051778ca8fb0a
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2022-09-22 09:56:47
# Regressor Bug: 1789967
# File Overlap Count: 1
# Description:
#   Bug 1789967 - part 2: Make `TextEditor` and `HTMLEditor` implement `EditorBase::CollapseSelectionToEndOfLastLeafNode` by themselves r=m_kato
#   
#   It does different thing for `TextEditor` and `HTMLEditor`, and used almost
#   internally.  Therefore, it should be implemented in the sub classes and
#   we should name them better.
# ==============================================================================

diff -r cc2e8b7c548e -r cdcef4f3f635 editor/libeditor/EditorBase.h
--- a/editor/libeditor/EditorBase.h	Thu Sep 22 06:06:54 2022 +0000
+++ b/editor/libeditor/EditorBase.h	Thu Sep 22 06:17:36 2022 +0000
@@ -1921,12 +1921,6 @@
   }
 
   /**
-   * CollapseSelectionToEnd() collapses the selection to the last leaf content
-   * of the editor.
-   */
-  MOZ_CAN_RUN_SCRIPT nsresult CollapseSelectionToEndOfLastLeafNode() const;
-
-  /**
    * AllowsTransactionsToChangeSelection() returns true if editor allows any
    * transactions to change Selection.  Otherwise, transactions shouldn't
    * change Selection.